sdk/resource/builtin.go

```diff
@@ -0,0 +1,49 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"os"
+
+	opentelemetry "go.opentelemetry.io/otel/sdk"
+	"go.opentelemetry.io/otel/semconv"
+)
+
+type (
+	TelemetrySDK struct{}
+	Host         struct{}
+)
+
+var (
+	_ Detector = TelemetrySDK{}
+	_ Detector = Host{}
+)
+
+func (TelemetrySDK) Detect(context.Context) (*Resource, error) {
+	return New(
+		semconv.TelemetrySDKNameKey.String("opentelemetry-go"),
+		semconv.TelemetrySDKLanguageKey.String("go"),
+		semconv.TelemetrySDKVersionKey.String(opentelemetry.Version()),
+	), nil
+}
+
+func (Host) Detect(context.Context) (*Resource, error) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
```

Comment:
nit: this should return a _partial_ error to not block startup.

Reply:
Maybe we should use `ErrorHandler`?

Reply:
I added tests for error handling, but had misunderstood _partial_ error support. I would say that a new issue should be filed about `ErrorHandler` support in `resource.Detect()`. That code does not call the global error handler, but I think it should.

While we're at it, we might consider improving `resource.Detect()` not to block startup indefinitely, with a context timeout, and to run detectors in parallel.